### DISCUSSION TITLE
Unable to build petsc

### DISCUSSION THREAD

**Discussion by sarshia** [firedrakeproject/firedrake#3266] [Firedrake support] [answered]
Hello, I am trying to install firedrake and along the way I keep running into errors when building petsc. The most current output says:

Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 1907, in <module>
    install("petsc/")
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 1051, in install
    build_and_install_petsc()
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 1163, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 664, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/arshiasinghal/Desktop/Rice/Research/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/arshiasinghal/Desktop/Rice/Research/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-ptscotch', '--CFLAGS=-Wno-implicit-function-declaration', '--download-pnetcdf', '--with-c2html=0', '--download-mumps', '--with-debugging=0', '--download-pastix', '--download-hwloc-configure-arguments=--disable-opencl', '--download-hdf5', '--download-mpich-configure-arguments=--disable-opencl', '--download-mpich', "--download-openblas-make-options='USE_THREAD=0 USE_LOCKING=1 USE_OPENMP=0'", '--with-zlib', '--download-scalapack', '--download-netcdf', '--download-hwloc', '--download-superlu_dist', '--download-hypre', '--with-x=0', '--download-suitesparse', '--download-openblas', '--with-shared-libraries=1', '--download-chaco', '--download-bison', '--with-fortran-bindings=0', '--download-metis']' returned non-zero exit status 1.

Here are the [firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/13478539/firedrake-install.log) and petsc [configure.log](https://github.com/firedrakeproject/firedrake/files/13478547/configure.log). I was previously getting an issue with gfortran so I deactivated conda, and I now get this issue instead. Please advise.

**Comment by connorjward** [comment]
@Ig-dolci is the most knowledgeable about current MacOS install issues. Does this look like the same Xcode issues that we've been seeing?

I do notice that you are not using a homebrew-installed Python which is not supported. Please install Python via homebrew and use this.

**Comment by sarshia** [comment]
Thank you for your response. I think Python was installed but unlinked so I linked it and tried to redo the installation. However, I'm not sure if this ensured that the homebrew-installed Python is being used since the python filepath in the new error message looks the same: 

Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 1907, in <module>
    install("petsc/")
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 1051, in install
    build_and_install_petsc()
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 1163, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/Users/arshiasinghal/Desktop/Rice/Research/firedrake-install", line 664, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/arshiasinghal/Desktop/Rice/Research/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/arshiasinghal/Desktop/Rice/Research/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-hwloc-configure-arguments=--disable-opencl', '--download-mpich', '--with-debugging=0', '--download-hypre', '--download-pastix', '--download-openblas', '--with-shared-libraries=1', '--download-mpich-configure-arguments=--disable-opencl', '--download-chaco', '--download-suitesparse', '--with-fortran-bindings=0', '--with-c2html=0', '--with-x=0', '--download-hdf5', '--download-superlu_dist', '--download-pnetcdf', "--download-openblas-make-options='USE_THREAD=0 USE_LOCKING=1 USE_OPENMP=0'", '--CFLAGS=-Wno-implicit-function-declaration', '--download-bison', '--download-netcdf', '--download-mumps', '--download-scalapack', '--with-zlib', '--download-metis', '--download-ptscotch', '--download-hwloc']' returned non-zero exit status 1.

Do you know how to check if the homebrew-installed Python is being used? Or should it perhaps be installed in another location?

**Reply by connorjward** [reply]
I think the best way to make sure you're using the right python is to run `$(brew --prefix)/bin/python3 firedrake-install`.

However, I don't think that this will resolve your current issue. If you are using an up to date MacOS you may need to install a beta version of Xcode. This is the main failure mode that we're seeing a lot of Mac users experience currently. Instructions to fix are [here](https://github.com/firedrakeproject/firedrake/wiki/Installation-Issues-with-Firedrake-on-Xcode-15-for-MacOS-Users).

**Comment by Ig-dolci** [comment]
> @Ig-dolci is the most knowledgeable about current MacOS install issues. Does this look like the same Xcode issues that we've been seeing?
> 
> I do notice that you are not using a homebrew-installed Python which is not supported. Please install Python via homebrew and use this.

This error is not related to the Xcode 15 version. As you said @connorjward, the right homebrew Python is not used here. 
I think that the @connorjward instructions can help you @sarshia.

**Reply by Ig-dolci** [reply]
@sarshia I saw a message about using Xcode 14.2. So, you can use it. We have noticed Xcode install problems for macOS users and with Xcode 15. Xcode 14 should work.

**Reply by connorjward** [reply]
Yes. Also looking at the log it seems like you hit a network error. This is likely to either have been caused by using the wrong Python (default MacOS Python has dodgy network configuration) or by dropping network connection part way through the install.

**Comment by sarshia** [comment]
Thank you both for your responses. I'm trying to download Xcode but it looks like it requires a lot of free disk space so I'll try to find a way to do that. I'll follow up once I have a chance to try it out.

**Reply by connorjward** [reply, marked as answer]
I don't think that the issue is with Xcode in your case. Please just try to use the correct Python and see what happens.

**Reply by sarshia** [reply]
I used the correct Python as you suggested and it worked! Thank you both again for your help.
